lib/recurly/pricing/calculations.js

```diff
@@ -128,8 +128,34 @@ Calculations.prototype.tax = function (done) {
  */
 
 Calculations.prototype.total = function () {
-  this.price.now.total = this.price.now.subtotal + this.price.now.tax;
-  this.price.next.total = this.price.next.subtotal + this.price.next.tax;
+  let nowTotal = this.price.now.subtotal + this.price.now.tax;
+  let nextTotal = this.price.next.subtotal + this.price.next.tax;
+
+  if(this.pricing.items.giftcard){
+    let giftcardNow = useGiftcard(nowTotal,(this.pricing.items.giftcard.unit_amount_in_cents/100));
+    let giftcardNext = useGiftcard(nextTotal,giftcardNow.remains);
+
+    this.price.now.giftcard = giftcardNow.used;
+    this.price.next.giftcard = giftcardNext.used;
+
+    nowTotal = nowTotal - giftcardNow.used;
+    nextTotal = nextTotal - giftcardNext.used;
+  }
+
+  this.price.now.total = nowTotal;
+  this.price.next.total = nextTotal;
+
+  function useGiftcard(price,giftcardValue){
+    let used = 0;
+    let remains = 0;
+    if(giftcardValue > price){
+      used = price;
+      remains = giftcardValue - price;
+    } else {
+      used = giftcardValue;
+    }
+    return { used, remains };
+  }
```

Comment:
Let's move gift card calculations into its own method. Total should be concerned only with figuring out the total price.